models/Notification.php

```diff
@@ -212,39 +212,39 @@ public function setSender(?User $sender): self
     }
 
     /**
-     * @return User|null
+     * @return User
```

Comment:
Why it cannot be null? If I create a new Notification it is null, right?
```
$notification = new Notification();
$notification->getRecipient();
```
In addition, the return type hint is still nullable

Reply:
@blankse Thanks, I will change the typehint.  It cannot be null because it doesn't make sense to create/send a new notification with an empty recipient.

Reply:
Yes, but I think that should check the save method. Until the setRecipient() method is called, the attribute is still null. So why shouldn't the user be able to use getRecipient to check whether it's already set?

Reply:
@blankse I have added the null check before the save() method. But, we decided to follow an approach where the  properties type would match with the db column types. That's why we made it to not null.

Reply:
For setter typehints I agree. But getter/properties can be null for new unsaved models. When you really want to change this its a BC break I think.

Reply:
yes, you are right. Reverted the changes for BC reasons. thanks!